### DISCUSSION TITLE
Way to passa properties to layout in next 13 (appDir)

### DISCUSSION THREAD

**Discussion by volnei** [vercel/next.js#42306] [Help] [answered]
There any way to pass properties or access children property from layout in next 13 (appDir)

**Comment by erikbg7** [comment, marked as answer]
Any layout file will receive the page as `children`:
https://nextjs.org/blog/next-13#layouts

So you should be able to modify, enhance and change the children props as you can do with React.
https://sho-mukai.medium.com/react-different-ways-to-enhance-child-components-113c1f54713c

**Comment by voyagebagage** [comment]
I have the same problem, but I only need to pass a string.
So if you don't have a function to pass, in that case what about: 
[**cookies**][1] or [**local storage**][2] ?


  [1]: https://www.npmjs.com/package/js-cookie
  [2]: https://blog.logrocket.com/using-localstorage-react-hooks/

**Comment by khaledseleem0** [comment]
> There any way to pass properties or access children property from layout in next 13 (appDir)

Does anyone have a specific tutorial because it does not work for me?

**Comment by ViniciusVC** [comment]
It took a lot of work to switch to Next13's default layout. And now I'm going to have to go back to my layout component.
